pkg/producer/kafka.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020 VMware, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package producer
+
+import (
+	"fmt"
+
+	"github.com/Shopify/sarama"
+	"github.com/golang/protobuf/proto"
+	"k8s.io/klog"
+
+	"github.com/vmware/go-ipfix/pkg/entities"
+	"github.com/vmware/go-ipfix/pkg/producer/protobuf"
+)
+
+var (
+	KafkaConfigVersion sarama.KafkaVersion
+	// ProtoSchemaTypes lists all proto schema supported in Kafka producer.
+	ProtoSchemaTypes = map[string]string{
+		"FlowType1": "FlowType1",
+		"FlowType2": "FlowType2",
+	}
```

Comment:
Just curious, do you really need the value part since it is the same as the key?
If not, maybe we can store the mapping as map[string]convertRecordToFlowType?
For example,

```
	ProtoSchemaTypeConverterTable = map[string]sconvertRecordToFlowType{
		"FlowType1": convertRecordToFlowType1,
		"FlowType2": convertRecordToFlowType2,
	}
```
	
Thus lines 52-57 can be simplified?

Reply:
Good idea.

Reply:
+1